lib/pbench/test/unit/server/test_user_auth.py

```diff
@@ -0,0 +1,390 @@
+
+    @staticmethod
+    def test_user_relogin(client, server_config):
+        """ Test for login of registered-user login """
+        with client:
+            # user registration
+            resp_register = register_user(
+                client,
+                server_config,
+                username="user",
+                firstname="firstname",
+                lastname="lastName",
+                email="user@domain.com",
+                password="12345",
+            )
+            data_register = resp_register.json
+            assert data_register["status"] == "success"
+            # registered user login
+            response = login_user(client, server_config, "user", "12345")
+            data = response.json
+            assert data["status"] == "success"
+            assert data["message"] == "Successfully logged in."
+            assert data["auth_token"]
+            assert response.content_type == "application/json"
+            assert response.status_code == 200
+
+            # Immediate re-login with auth header
+            response = client.post(
+                f"{server_config.rest_uri}/session",
+                headers=dict(Authorization="Bearer " + data["auth_token"]),
+                json={"username": "user", "password": "12345"},
+            )
+            data = response.json
+            assert data["message"] == "Retry login after some time"
+            assert response.status_code == 403
+
+            # Immediate re-login without auth header
+            response = client.post(
+                f"{server_config.rest_uri}/session",
+                json={"username": "user", "password": "12345"},
+            )
+            data = response.json
+            assert data["message"] == "Retry login after some time"
+            assert response.status_code == 409
+
+            # Re-login after a second
+            time.sleep(1)
+            response = client.post(
+                f"{server_config.rest_uri}/session",
+                json={"username": "user", "password": "12345"},
+            )
+            data = response.json
+            assert data["status"] == "success"
+
+    @staticmethod
+    def test_user_login_with_wrong_password(client, server_config):
+        """ Test for login of registered-user login """
+        with client:
+            # user registration
+            resp_register = register_user(
+                client,
+                server_config,
+                username="user",
+                firstname="firstname",
+                lastname="lastName",
+                email="user@domain.com",
+                password="12345",
+            )
+            data_register = resp_register.json
+            assert data_register["status"] == "success"
+            # registered user login
+            response = login_user(client, server_config, "user", "123456")
+            data = response.json
+            assert data["message"] == "Wrong password"
```

Comment:
I guess I didn't even pay attention to this when I was reading the actual code; but this could be considered a security problem. You're effectively telling an attacker they guessed at a real username but got the password wrong. The general best practice is to give no clue they got _anything_ right. Maybe we should just say "Bad login" or "Better luck next time". ;-)